### ISSUE TITLE
Validation webhook broken by v1beta1 CRD update

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#20449]
Validation logs are spamming `unrecognized type Sidecar`. Adding some more debug logs shows: `unrecognized type {{Sidecar networking.istio.io/v1beta1}`.

Pilot only has schemas for v1alpha3, so when we give it a v1beta it explodes.

This breaks all validation.

The weird thing is I don't even think I am applying these sidecars, they are just already existing in my cluster

**Comment by ayj** [comment]
I initially saw the same problem. I'm not sure why we'd see that in the steady state. The root cause is that we've registered the validatingwebhookconfiguration for all verions but we don't support those versions yet in pilot/galley. Restricting validation to a specific version won't work. Other versions would skip validation. I think the fix here is to update the metadata.yaml to include both versions. 

We can continue to use the v1alphaX or switch to v1beta1. The latter will be a fairly large PR.

**Comment by howardjohn** [comment]
I am still concerned by the steady state. Is kubernetes calling the validation to convert already stored resources or something?

I don't think we should make pilot internally use beta api, when the version bump was merged the understanding was that we would not require changes to pilot

**Comment by ayj** [comment]
Did this problem happen after an update? We do apply some initial configuration such as the default sidecar in istio-system. I suppose its possible the Istio API was applied after istiod registered the webhook config.

**Comment by howardjohn** [comment]
I had a cluster that I had continually updated from master with istiod. I never had galley/validation until your changes landed, but the Sidecars I applied are part of the standard perf test setup so I am fairly sure they are valid. When I upgraded to some change that had the validation+v1beta1 update, I started seeing this. I am 90% sure it was for Sidecars that I had applied, not (just) the istio-system one, but my memory could be wrong.
